Pass numbers to the front end as strings

kamil4 opened the pull request:
Fixes https://github.com/LycheeOrg/Lychee-front/issues/161

The issue there was the identity operator (`===`) resolving to `false` when one side was an integer and the other was a string. It could happen because we were being sloppy on the server side, especially with IDs, sometimes returning them as integers, sometimes as strings. This PR attempts to fix up all such cases that I could find, turning them into strings, which as a bonus should avoid any potential 32-bit issues as well.

ildyria commented on app/Http/Controllers/AlbumController.php:
```diff
@@ -155,10 +155,10 @@ public function get(Request $request)
 
 		$return['id'] = $request['albumID'];
-		$return['num'] = count($return['photos']);
+		$return['num'] = strval(count($return['photos']));
```
why ? Semantically this one does not make sense.

ildyria commented on app/Http/Controllers/AlbumController.php:
```diff
@@ -155,10 +155,10 @@ public function get(Request $request)
 
 		// finalize the loop
-		if ($return['num'] === 0) {
+		if ($return['num'] === '0') {
```
Idem

ildyria commented on app/ModelFunctions/AlbumFunctions.php:
```diff
@@ -342,7 +342,7 @@ public function genSmartAlbumsThumbs(array &$return, Builder $photos_sql, string
 			'thumbs2x' => array(),
 			'types' => array(),
-			'num' => $photos_sql->count(),
+			'num' => strval($photos_sql->count()),
```
item.

ildyria commented on app/Photo.php:
```diff
@@ -170,13 +170,13 @@ public function prepareData()
-		$photo['width'] = $this->width;
-		$photo['height'] = $this->height;
+		$photo['album'] = $this->album_id !== null ? strval($this->album_id) : null;
+		$photo['width'] = strval($this->width);
```
given that they are used as strings in the front-end sure. :)
Does this interfere with the Justified Layout ??

ildyria commented on app/Photo.php:
```diff
@@ -170,13 +170,13 @@ public function prepareData()
-		$photo['height'] = $this->height;
+		$photo['album'] = $this->album_id !== null ? strval($this->album_id) : null;
+		$photo['width'] = strval($this->width);
+		$photo['height'] = strval($this->height);
```
given that they are used as strings in the front-end sure. :)
Does this interfere with the Justified Layout ??

d7415 commented:
I'm mostly a fan of treating numbers as numbers, but consistency is important, and it's not like we use ID as an int once it's generated. It also removes any requirement for special handling of leading zeroes.

I agree with @ildyria though, that anything used as a number in the front end should really be left as such unless we have a good reason.

kamil4 commented:
You are correct that only the integer IDs were found to be a problem in practice so I could've stopped there.

I converted `width` and `height` to strings as well because that's how they are sent in v3 and I figured it was better to be consistent (they were the only int fields in their JSON objects; everything else seems to be sent as strings, including 0/1 flags, ISO values, latitude/longitude, etc.). So, yes, they work fine as strings with the different layouts that we have.

On closer inspection, v3 sends `num` as an integer just like v4, so I may have been overzealous there :wink:. I converted it to a string because it was the last remaining non-string field in the whole response so, again, I figured it was better to be consistent. It doesn't seem to cause any problems.

Let me know if this response changed your minds. I have no problem reverting the non-essential changes, of course.

d7415 commented:
Consistency with v3 is a good enough reason for me. If we ever release v4 we could revisit some day.

ildyria commented:
If this is consistent with v3, then it is fine. However, once the v4 is out and we drop the v3, I think it would be nice to double check everything. :)
